spec/models/ineligibility_reason_checker_spec.rb

```diff
@@ -0,0 +1,557 @@
+require "rails_helper"
+
+RSpec.describe IneligibilityReasonChecker do
+  let(:academic_year) { AcademicYear.new(2022) }
+  let(:none_of_the_above_academic_year) { AcademicYear::Type.new.serialize(AcademicYear.new("None")) }
+  let(:ecp_claim) { build(:claim, policy: Policies::EarlyCareerPayments, academic_year:, eligibility: ecp_eligibility, logged_in_with_tid:, qualifications_details_check:) }
+  let(:lup_claim) { build(:claim, policy: Policies::LevellingUpPremiumPayments, academic_year:, eligibility: lup_eligibility, logged_in_with_tid:, qualifications_details_check:) }
+  let(:logged_in_with_tid) { nil }
+  let(:qualifications_details_check) { nil }
+
+  let(:school_eligible_for_ecp_and_lup) { create(:school, :early_career_payments_eligible, :levelling_up_premium_payments_eligible) }
+  let(:school_eligible_for_ecp_but_not_lup) { create(:school, :early_career_payments_eligible) }
+  let(:school_eligible_for_lup) { create(:school, :levelling_up_premium_payments_eligible) }
+  let(:school_ineligible_for_both_ecp_and_lup) { create(:school, :early_career_payments_ineligible) }
+
+  before { create(:journey_configuration, :additional_payments, current_academic_year: academic_year) }
+
+  # sanity check of factories
+  specify { expect(Policies::EarlyCareerPayments::SchoolEligibility.new(school_eligible_for_ecp_but_not_lup)).to be_eligible }
+  specify { expect(Policies::LevellingUpPremiumPayments::SchoolEligibility.new(school_eligible_for_ecp_but_not_lup)).not_to be_eligible }
+
+  specify { expect(Policies::EarlyCareerPayments::SchoolEligibility.new(school_ineligible_for_both_ecp_and_lup)).not_to be_eligible }
+  specify { expect(Policies::LevellingUpPremiumPayments::SchoolEligibility.new(school_ineligible_for_both_ecp_and_lup)).not_to be_eligible }
+
+  let(:current_claim) { CurrentClaim.new(claims: [ecp_claim, lup_claim]) }
+
+  let(:journey) { "additional-payments" }
+  let(:answers) { Journeys::SessionAnswers.new }
+  let(:journey_session) do
+    object = Journeys::AdditionalPaymentsForTeaching::Session.new(journey:)
+    object.load_answers(answers)
+    object
+  end
```

Comment:
I think the way to go with this is setting up some factories for the answers for each scenario
https://github.com/DFE-Digital/claim-additional-payments-for-teaching/blob/CAPT-1650/itt-year-form/spec/lib/ineligibility_reason_checker_spec.rb

Reply:
I originally tried but ran into some issues (don't recall the failures), but i can have another play

Reply:
I'm now using the factories as you've suggested and `#load_answers` is no longer needed and has been removed